### PR TITLE
Avoid /dev/null and NUL in other.test*respect*: it is irrelevant, and hits odd windows behavior

### PR DESCRIPTION
**Pull request by kripken**
It fails on python3 for some reason on chromium CI, but isn't important anyhow.

### CODE REVIEW

File: tests/test_other.py [hunk 1]
```diff
@@ -4578,11 +4578,7 @@ def test(args=[]):
     test(['--bind'])
 
   def test_dashE_respect_dashO(self):
-    # issue #3365
```
Comment (dschuff) [inline tests/test_other.py]: should the issue link be removed?

Comment (kripken) [reply tests/test_other.py]: Oh, heh, that's bad. I'll fix before I land.

Comment (dschuff) [reply tests/test_other.py]: Although, it's not obvious what the issue has to do with this test.

**Comment by juj** [comment]
What was the odd Windows behavior? I think I had added this test because people used `> /dev/null`/`NUL` routing with Emscripten (on some build systems?)

File: tests/test_other.py (hunk 1 above)
Comment (juj) [reply tests/test_other.py]: Looks like the #issue 3365 comments were left around in the landed form.

Comment (kripken) [reply tests/test_other.py]: Yes, I preserved the history, as those tests were added because of that issue.

Comment (kripken) [reply tests/test_other.py]: (That might have been a mistake at the time, I didn't investigate now...)

**Comment by kripken** [comment]
The odd thing was that a file called `NUL` was created on windows, so that assertion failed (!).

I believe the test checks for `-o` behavior, regardless of the actual file specified. If we think it's important to specifically test `/dev/null` / `NUL` behavior, I can look into adding that separately?

**Comment by juj** [comment]
> The odd thing was that a file called `NUL` was created on windows

That should not be possible in any way, Windows FAT32 and NTFS file system implementation prevent a file or directory named `NUL` from being created on the filesystem level. Trying e.g. `mkdir NUL`, `echo asdf > NUL` or `copy emcc.py NUL` will all succeed, but will not create a file. Trying locally, any of

```
emcc tests\hello_world.c -o NUL
em++ tests\hello_world.cpp -E -o NUL
em++ tests\hello_world.cpp -M -o NUL
```
succeeds, but no file named `NUL` is created. Also testing
```py
import os
print(str(os.path.exists('NUL')))
```
yields `False` at least on Python 2.7.13 and Python 3.7.2 on Windows.

I would rather be suspecting a Python bug if `self.assertNotExists('NUL')` is suddenly failing. (Also this test has existed for a long time, I wonder why it would start failing all of a sudden?)

**Comment by kripken** [comment]
The change was on chromium CI - some updates happened there, which broke a bunch of things, and over the week we moved to python3 on windows as part of the solution. The problem started to appear after that change. We can't reproduce the issue on local windows machines, very strange...

Note that the problem might not be that it successfully creates a file called `NUL`, but maybe it always says such a file exists on that machine? Might be worth verifying that if it would matter.

**Comment by juj** [comment]
> the problem might not be that it successfully creates a file called `NUL`, but maybe it always says such a file exists on that machine? Might be worth verifying that if it would matter.

That is what I tried verifying above with `print(str(os.path.exists('NUL')))` - that does yield `False` on my Windows computer at least on those python versions. Not sure if a python version difference might have an effect here..

**Comment by kripken** [comment]
Yeah, it's very odd.. Anyhow, I agree that testing `-o /dev/null` is worth keeping. But I don't think verifying that the output doesn't exist makes sense (as that's the OS's business). I opened #10037 with a new test for that.

**Pull request merged**